src/Application/Actions/RegularGivingMandate/Get.php

```diff
@@ -0,0 +1,47 @@
+<?php
+declare(strict_types=1);
+namespace App\Application\Actions\RegularGivingMandate;
+use JetBrains\PhpStorm\Pure;
+use Laminas\Diactoros\Response\JsonResponse;
+use MatchBot\Application\Actions\Action;
+use MatchBot\Application\Auth\PersonWithPasswordAuthMiddleware;
+use MatchBot\Domain\DomainException\DomainRecordNotFoundException;
+use MatchBot\Domain\PersonId;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Log\LoggerInterface;
+use Slim\Exception\HttpNotFoundException;
+
+class Get extends Action {
+
+    #[Pure]
+    public function __construct(
+        LoggerInterface $logger
+    ) {
+        parent::__construct($logger);
+    }
+
+    protected function action(Request $request, Response $response, array $args): Response
+    {
+        if (! $this->environment->isFeatureEnabledRegularGiving()) {
+            throw new HttpNotFoundException($request);
+        }
+        if (empty($args['mandateId'])) {
+            throw new DomainRecordNotFoundException('Missing donation ID');
```

Comment:
Liking the validation but I think its worth avoiding  `empty` in general. We could add a rule against empty to psalm config. See https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/

Probably fine not to do any particular check here actually, fine to rely on the throw when the regularGivingMandateRepository doesn't return anything.

Reply:
not a blocker, may try and remove usage of empty more generally at some point.